src/main/java/com/joy/joyfillandroid/Tablefield/MobileSDKField.kt

```diff
@@ -0,0 +1,401 @@
+                            ShowComponents(Sequence_With_Y_Position, fields)
+                        }
+                    }else {
+                        // For Primary View or Web View
+                        for (j in 0 until pages.length()) {
+                            val fieldPositions = pages.getJSONObject(j).getJSONArray("fieldPositions")
+                            for (k in 0 until fieldPositions.length()) {
+                                val component_Type = fieldPositions.getJSONObject(k).get("type")
+                                val Y_Position =  fieldPositions.getJSONObject(k).getDouble("y")
+                                yValues_mobile.add(Y_Position.toInt())
+                                component_type.add(component_Type.toString())
+                                val componentTypePairedArray = yValues_mobile.zip(component_type)
+                                val sorted_type = componentTypePairedArray.sortedBy { it.first }
+                                val sortedType_value = valueForSortedType(sorted_type.toString())
+                                Sequence_With_Y_Position = sortedType_value
+                                dataValues.addAll(valueForSortedType(sorted_type.toString()))
+                            }
+                            formatBlockData(fieldPositions)
+
+                        }
+                        ShowComponents(Sequence_With_Y_Position, fields)
+                    }
+                }
+                updateVaue(fields)
+                getTableData(fields)
+            }
+
+            override fun onFailure(call: okhttp3.Call, e: IOException) {
+                println("call "+ call)
+            }
+        })
+    }
+
+    fun getTableData(fields: JSONArray){
+        for (i in 0 until fields.length()) {
+            val type = fields.getJSONObject(i).get("type")
+
+            if (type == "table"){
+                val tableColumnsArray = fields.getJSONObject(i).getJSONArray("tableColumns")
+                val rowOrderArray = fields.getJSONObject(i).getJSONArray("rowOrder")
+                val valueArray = fields.getJSONObject(i).getJSONArray("value")
+                val tableColumnOrderArray = fields.getJSONObject(i).getJSONArray("tableColumnOrder")
+                var orderID: String? = null
+                for (i in 0 until tableColumnOrderArray.length()) {
+                    orderID = tableColumnOrderArray.getString(i)
+                    tableColumnOrderArrayID.add(orderID)
+                }
+
+                val tableColumnID =  fields.getJSONObject(i).getJSONArray("tableColumns")
+                for (i in 0 until tableColumnID.length()) {
+                    val columnId = tableColumnID.getJSONObject(i).optString("_id")
+                    idPositionsMap[columnId] = i
+                }
+
+                for (j in 0 until tableColumnOrderArrayID.size){
+                    val idFromShortedData = tableColumnOrderArrayID[j]
```

Comment:
So here is a good example to look at all code in here to redo or moving forward where I know at teams it takes more effort to name things nicely but it's important for other devs coming in. Here you have `idFromShortedData` that really doesn't give me any info other than I think this results in an id.

Reply:
This idFromShortedData is related to get table column id’s but you are right it is difficult to understand this with that name so I renamed it with getTableColumnOrderId.